wandb/integration/openai/fine_tune.py

```diff
@@ -0,0 +1,350 @@
+        :param force: Forces logging and overwrite existing wandb run of the same fine-tune.
+        :param blocking: Waits for the fine-tune to be complete and then log metrics to W&B. By default, it is True.
+        """
+
+        openai_client = OpenAI(
+            api_key=os.environ['OPENAI_API_KEY'],
+        )
+        cls.openai_client = openai_client
+
+        if id:
+            print("Retrieving fine-tune job...")
+            fine_tune = openai_client.fine_tuning.jobs.retrieve(fine_tuning_job_id=id)
+            fine_tunes = [fine_tune]
+        else:
+            # get list of fine_tune to log
+            fine_tunes = openai_client.fine_tuning.jobs.list()
+            if not fine_tunes or fine_tunes.data is None:
+                print("No fine-tune has been retrieved")
+                return
+            fine_tunes = fine_tunes.data[
+                -n_fine_tunes if n_fine_tunes is not None else None :
+            ]
+
+        # log starting from oldest fine_tune
+        show_individual_warnings = (
+            False if id is None and n_fine_tunes is None else True
+        )
+        fine_tune_logged = []
+        for fine_tune in fine_tunes:
+            if blocking:
+                fine_tune = cls._wait_for_job_success(fine_tune)
+
+            cls._log_fine_tune(
+                fine_tune,
+                project,
+                entity,
+                force,
+                show_individual_warnings,
+                **kwargs_wandb_init,
+            )
+
+        if not show_individual_warnings and not any(fine_tune_logged):
+            print("No new successful fine-tunes were found")
+
+        return "🎉 wandb sync completed successfully"
+
+    @classmethod
+    def _wait_for_job_success(cls, fine_tune):
+        wandb.termwarn("Waiting for the Fine-tuning job to be finished...")
+        while True:
+            if fine_tune.status == "succeeded":
+                wandb.termwarn("Fine-tuning finished, logging metrics to W&B")
+                return fine_tune
+            if fine_tune.status == "failed":
+                wandb.termwarn(
+                    "Fine-tune {fine_tune_id} has has failed and will not be logged"
+                )
+                return fine_tune
+            if fine_tune.status == "cancelled":
+                wandb.termwarn(
+                    "Fine-tune {fine_tune_id} has was cancelled and will not be logged"
+                )
+                return fine_tune
+            time.sleep(60)
+            fine_tune = cls.openai_client.fine_tuning.jobs.retrieve(fine_tuning_job_id=fine_tune.id)
+
+    @classmethod
+    def _log_fine_tune(
+        cls,
+        fine_tune,
+        project,
+        entity,
+        force,
+        show_individual_warnings=None,
+        **kwargs_wandb_init,
+    ):
+        fine_tune_id = fine_tune.id
+        status = fine_tune.status
+
+        # check run completed successfully
+        if status != "succeeded":
```

Comment:
If the status is not "succeeded" it will `return` without logging anything and will display this warning message: `f'Fine-tune {fine_tune_id} has the status "{status}" and will not be logged'`.

However, this warning should not be encountered by most users because of `wait_for_job_success` is True by default.